src/domain/metadata/domain.ts

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable max-lines-per-function */
+import { createStore } from 'zustand'
+import type { StoreApi } from 'zustand'
+import { devtools } from 'zustand/middleware'
+import { immer } from 'zustand/middleware/immer'
+import * as O from 'fp-ts/Option'
+import * as A from 'fp-ts/Array'
+import * as RTE from 'fp-ts/ReaderTaskEither'
+import { flow, pipe } from 'fp-ts/function'
+import type { ForgetType } from '@/util/type'
+import type { AuditMetadataItem, GeneralMetadataItem, Metadata, MetadataItem } from './entity'
+import type { Query } from './query'
+import type { Command, Deps } from './command'
+import { isDevMode } from '@/util/env.util'
+import { isNetworkRequestAbortedError } from '@/shared/error/network'
+
+type State = {
+  data: Metadata
+}
+
+type Domain = State & Command & Query
+export type DomainAPI = ForgetType<State, Domain>
+
+export type Options = {
+  initialState: State
+}
+const isAuditMetadataItem = (metadataItem: MetadataItem): metadataItem is AuditMetadataItem =>
+  metadataItem.category === 'auditMetadata'
+
+const isGeneralMetadataItem = (metadataItem: MetadataItem): metadataItem is GeneralMetadataItem =>
+  metadataItem.category === 'generalMetadata'
+
+export const storeFactory = ({ initialState }: Partial<Options> = {}): StoreApi<DomainAPI> =>
+  createStore(
+    devtools(
+      immer<Domain>((set, get) => ({
+        data: pipe(
+          initialState,
+          O.fromNullable,
+          O.map(it => it.data),
+          O.getOrElse<Metadata>(() => ({
+            data: [],
+            isLoading: false
+          }))
+        ),
+        isLoading: () => () => get().data.isLoading,
+        auditMetadata: () => () =>
+          pipe(
+            get().data.data,
+            A.filter(isAuditMetadataItem),
+            A.map(auditMetadata => ({
+              property: auditMetadata.property,
+              value: auditMetadata.value
+            }))
+          ),
+        generalMetadata: () => () =>
+          pipe(
+            get().data.data,
+            A.filter(isGeneralMetadataItem),
+            A.map(generalMetadata => ({
+              property: generalMetadata.property,
+              value: generalMetadata.value
+            }))
+          ),
```

Comment:
To avoid code repetition, we could have a single `metadataByType` function that return metadata regarding the type parameter.

Reply:
I agree with you 100%. I suggest you do a refactor PR as a 2nd step to improve all the points raised. 
Is that ok with you @fredericvilcot ?

Reply:
Sure 😌